File: Python/AP-1/words_without.py

# given an array of strings, return a new array without the strings that are equal to the target string
def words_without(words, target):
    new_list = []

    i = 0
    while i < len(words):
        if words[i] != target:
            new_list.append(words[i])
        i += 1

    return new_list

print(words_without(['a', 'b', 'c', 'a'], 'a'))
print(words_without(['a', 'b', 'c', 'a'], 'b'))
print(words_without(['a', 'b', 'c', 'a'], 'c'))

File: JavaScript/AP-1/wordsWithout.js

// given an array of strings, return a new array without the strings that are equal to the target string
// one approach is to count the occurrences of the target string,
// make a new array of the correct length, and then copy over the correct strings
const wordsWithout = (words, target) => {
	let newArray = [];

	for (let i = 0; i < words.length; i++) {
		if (words[i] !== target) {
			newArray.push(words[i]);
		}
	}

	return newArray;
};

console.log(wordsWithout(['a', 'b', 'c', 'a'], 'a'));
console.log(wordsWithout(['a', 'b', 'c', 'a'], 'b'));
console.log(wordsWithout(['a', 'b', 'c', 'a'], 'c'));
